soc/nxp/kinetis/Kconfig.defconfig

```diff
@@ -3,14 +3,6 @@
 
 if SOC_FAMILY_KINETIS
 
-config SERIAL_INIT_PRIORITY
-	default 55
-	depends on SERIAL
```

Comment:
The root of the override of SERIAL is from this PR #39452 to fix a problem that someone found with the order of initialization between SERIAL and the CLOCK control drivers. The root problem was that both classes of drivers had the same priority and by a somewhat undefined linker behavior the initialization order just happened to work until someone built with a different build environment and the linker decided to order things in a reverse order. The referenced PR created the generic config for SERIAL and with a separate commit the NXP parts were forced to this priority.

We would need to make sure that this doesn't get the same initialization priority as the clock control driver before removing this. 

I also note that I had a draft PR that was pulled in favor of the above PR and we appear to have missed pulling over the comment block that was above this code explaining the reasoning for the override.

Reply:
the clock control has priority 30 by default, serial has priority 50 by default. so changing it to 55 doesn't seem necessary.

Reply:
At the time of the change they were in the same priority. I think there is additional activity in the project to break this type of coding of priority and leverage something in device tree to configure dependencies.

Reply:
the init priorities are a topic in Arch WG right now, as far as this PR, just want to remove these overrides which are no longer relevant